model/generators/markets.py

```diff
@@ -120,41 +115,22 @@ def market_price(self, state):
         """
         # TODO check that slicing with step is correct
         step = state["timestep"]
-        if self.model == MarketPriceModel.GBM:
-            market_prices = {
-                "cusd_usd": (
-                    state["market_price"]["cusd_usd"]
-                    * self.increments["cusd_usd"][step - 1]
-                ),
-                "celo_usd": (
-                    state["market_price"]["celo_usd"]
-                    * self.increments["celo_usd"][step - 1]
-                ),
-            }
+        market_prices = {}
+        if self.model == MarketPriceModel.QUANTLIB:
+
+            for asset in state["market_price"]:
+                market_prices[asset] = (state["market_price"][asset]
+                                        * np.exp(self.increments[asset][step - 1]))
```

Comment:
why is it `step - 1` for QuantLib processes, and just `step` for the other models?